src/hotspot/share/gc/g1/g1CollectionSetCandidates.cpp

```diff
@@ -106,16 +121,12 @@ int G1CollectionCandidateList::compare(G1CollectionSetCandidateInfo* ci1, G1Coll
     return -1;
   }
 
-  double gc_eff1 = ci1->_gc_efficiency;
-  double gc_eff2 = ci2->_gc_efficiency;
+  size_t reclaimable1 = ci1->_r->reclaimable_bytes();
+  size_t reclaimable2 = ci2->_r->reclaimable_bytes();
 
-  if (gc_eff1 > gc_eff2) {
-    return -1;
-  } if (gc_eff1 < gc_eff2) {
-    return 1;
-  } else {
-    return 0;
-  }
+  if (reclaimable1 == reclaimable2) return 0;
+  if (reclaimable1 > reclaimable2) return -1;
+  return 1;
```

Comment:
Please use the same style for this if-elseif-else cascade than for the other one.